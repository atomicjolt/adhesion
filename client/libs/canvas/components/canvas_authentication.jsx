import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const select = state => ({
  settings: state.settings
});

export class CanvasAuthentication extends React.Component {
  static defaultProps = {
    overrides  : {},
    hideButton : false,
    autoSubmit : false,
  }

  static propTypes = {
    overrides  : PropTypes.shape({}),
    hideButton : PropTypes.bool,
    autoSubmit : PropTypes.bool,
    settings   : PropTypes.shape({
      canvas_oauth_url: PropTypes.string,
    }).isRequired,
  }

  componentDidMount() {
    if (this.props.autoSubmit) {
      this.form.submit();
    }
  }

  getButton() {
    if (this.props.hideButton) return null;
    return <input type="submit" value="Authorize" />;
  }

  renderSettings() {
    const settings = { ...this.props.settings, ...this.props.overrides };
    return _.map(settings, (value, key) => (
      <input key={key} type="hidden" value={value || ''} name={key} />
    ));
  }

  render() {
    return (
      <form
        ref={(ref) => { this.form = ref; }}
        action={this.props.settings.canvas_oauth_url}
      >
        { this.getButton() }
        { this.renderSettings() }
      </form>
    );
  }
}

export default connect(select)(CanvasAuthentication);
