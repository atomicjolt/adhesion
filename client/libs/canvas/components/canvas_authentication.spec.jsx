import React from 'react';
import { shallow } from 'enzyme';
import { CanvasAuthentication } from './canvas_authentication';

describe('the canvas authentication', () => {
  let props;
  let result;

  beforeEach(() => {
    props = {
      overrides: {},
      settings: {
        canvas_oauth_url: 'http://www.example.com',
      },
      hideButton: false,
      autoSubmit: false,
    };
    result = shallow(<CanvasAuthentication {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
