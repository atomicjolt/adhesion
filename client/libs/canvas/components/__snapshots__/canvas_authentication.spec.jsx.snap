// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the canvas authentication matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    action="http://www.example.com"
>
    <input
        type="submit"
        value="Authorize"
    />
    <input
        name="canvas_oauth_url"
        type="hidden"
        value="http://www.example.com"
    />
</form>,
  "nodes": Array [
    <form
      action="http://www.example.com"
>
      <input
            type="submit"
            value="Authorize"
      />
      <input
            name="canvas_oauth_url"
            type="hidden"
            value="http://www.example.com"
      />
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CanvasAuthentication
        autoSubmit={false}
        hideButton={false}
        overrides={Object {}}
        settings={
                Object {
                        "canvas_oauth_url": "http://www.example.com",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CanvasAuthentication {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "autoSubmit": false,
          "hideButton": false,
          "overrides": Object {},
          "settings": Object {
            "canvas_oauth_url": "http://www.example.com",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          action="http://www.example.com"
>
          <input
                    type="submit"
                    value="Authorize"
          />
          <input
                    name="canvas_oauth_url"
                    type="hidden"
                    value="http://www.example.com"
          />
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          action="http://www.example.com"
>
          <input
                    type="submit"
                    value="Authorize"
          />
          <input
                    name="canvas_oauth_url"
                    type="hidden"
                    value="http://www.example.com"
          />
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CanvasAuthentication
    autoSubmit={false}
    hideButton={false}
    overrides={Object {}}
    settings={
        Object {
            "canvas_oauth_url": "http://www.example.com",
          }
    }
/>,
}
`;
